### PR TITLE
[not ready] Implement match_arm_forces_newline option (#2039)

### PR DESCRIPTION
**Pull request by osa1**
This is an attempt at implementing #2039. I don't quite understand `rewrite_match_body` function yet, I just played around with some parameters until it indented as I wanted.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -622,6 +622,8 @@ create_config! {
     merge_derives: bool, true, "Merge multiple `#[derive(...)]` into a single one";
     binop_separator: SeparatorPlace, SeparatorPlace::Front,
         "Where to put a binary operator when a binary expression goes multiline.";
+    match_arm_forces_newline: bool, false,
```
Comment (nrc) [inline src/config.rs]: I would call this `match_arm_force_block`. The description should mention blocks rather than newlines too.

**Comment by osa1** [comment]
I think this PR was previously misunderstood -- I now added examples which hopefully make the intention here more clear. This flag when combined with `wrap_match_arms: false` does not change block-ness of an arm, but it consistently indents all arm bodies no matter how long the arm is or whether it was a block or not.

This style may seem weird to others (especially the empty block case) but I like this a lot because I know exactly where to look for the body when I see a match expression.

I understand if we don't want to merge this (I may be the only user of this). I kinda wish that rustfmt was more customizable and allowed to have this effect via some combination of existing flags. I think that's currently not possible. If anyone know a flag that's (1) more useful than that and (2) allows indenting match arms this way when combined with some other flags I'd be more than happy to implement that.

**Comment by nrc** [comment]
Thanks for adding tests! Could you add documentation of the option to https://github.com/rust-lang-nursery/rustfmt/blob/master/Configurations.md please? r+ with that.

**Comment by osa1** [comment]
Done! Thanks for the review. Let me know if you want me to rename the flag, it may be impossible to rename after publishing it.

**Comment by nrc** [comment]
Great, thanks!

**Pull request merged**